patches/tModLoader/Terraria/GameContent/Tile_Entities/TEDisplayDoll.cs.patch

```diff
@@ -0,0 +1,60 @@
+--- src/Terraria/Terraria/GameContent/Tile_Entities/TEDisplayDoll.cs
++++ src/tModLoader/Terraria/GameContent/Tile_Entities/TEDisplayDoll.cs
+@@ -4,11 +_,13 @@
+ using Terraria.Audio;
+ using Terraria.DataStructures;
+ using Terraria.GameInput;
++using Terraria.ModLoader;
++using Terraria.ModLoader.IO;
+ using Terraria.UI;
+ 
+ namespace Terraria.GameContent.Tile_Entities
+ {
+-	public class TEDisplayDoll : TileEntity
++	public sealed partial class TEDisplayDoll : TileEntity
+ 	{
+ 		private static byte _myEntityID;
+ 		private const int MyTileID = 470;
+@@ -105,18 +_,14 @@
+ 			for (int i = 0; i < 8; i++) {
+ 				Item item = _items[i];
+ 				if (!item.IsAir) {
+-					writer.Write((short)item.netID);
+-					writer.Write(item.prefix);
+-					writer.Write((short)item.stack);
++					ItemIO.Write(item, writer, !networkSend || ModNet.AllowVanillaClients, writeStack: true);
```

Comment:
See comments in Discord about increasing the item limit. The correct thing to do here is just `if (networkSend) ItemIO.SendModData(item, writer)`

Reply:
This is already correct. The write method writes mod data only if vanilla write boolean is false, and in this case it's only false if this is a network send with no vanilla compatibility.

Reply:
Reverted these patches and made OOP overrides for modded contexts.